### PR TITLE
[#ICC-97]  Add getThirdPartyMessage implementation

### PR DESCRIPTION
**Pull request by gquadrati**
This PR depends on https://github.com/pagopa/io-functions-app-messages/pull/25

#### List of Changes

- Add `getThirdPartyMessage` implementation
- Add tests

#### Motivation and Context

Enable Third Party messages

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -427,13 +429,20 @@ export function newApp({
           authMiddlewares.bearerSession,
           authMiddlewares.local
         );
-        // Create the messages service.
+
+        const thirdPartyClientFactory = getThirdPartyServiceClientFactory(
```
Comment (fabriziopapi) [inline src/app.ts:433]: This factory should use pnFetch if service is PN

File: src/services/newMessagesService.ts [hunk 2]
```diff
@@ -310,6 +360,69 @@ export default class NewMessagesService {
...
+              default:
+                return ResponseErrorStatusNotDefinedInSpec(response);
+            }
+          })
+        )
+      ),
+      TE.chainW(
+        // MessageWithThirdPartyData.is fails, we need to check the decode instead
+        TE.fromPredicate(isMessageWithThirdPartyData, () =>
+          ResponseErrorInternal(
```
Comment (AleDore) [inline src/services/newMessagesService.ts:419]: In this case we can return Bad Request instead

File: src/services/newMessagesService.ts [hunk 3]
```diff
@@ -340,6 +454,83 @@ export default class NewMessagesService {
...
+      TE.fromEither,
+      TE.mapLeft(error => {
+        log.error(
+          "newMessagesService|getThirdPartyMessageFromThirdPartyService|%s",
+          error.message
+        );
+        return ResponseErrorInternal(error.message);
+      }),
+      TE.map(getClientByFiscalCode =>
+        getClientByFiscalCode(message.fiscal_code)
```
Comment (AleDore) [inline src/services/newMessagesService.ts:480]: maybe we must pass serviceId here?

File: src/services/newMessagesService.ts (hunk 2 above)
Comment (gquadrati) [reply src/services/newMessagesService.ts:419]: Sure

File: src/app.ts (hunk 1 above)
Comment (gquadrati) [reply src/app.ts:433]: Sure

File: src/services/newMessagesService.ts (hunk 2 above)
Comment (gquadrati) [reply src/services/newMessagesService.ts]: Done in [78754ae](https://github.com/pagopa/io-backend/pull/899/commits/78754aee52cd910daf09259253fee5521f3792bd)

**Pull request merged**